### PR TITLE
Update Spine

### PR DESCRIPTION
**Pull request by dmdashenkov**
In this PR we:
 1. Update Spine versions. Now, the `spine/options.proto` contains the `compare_by` option. After ProtoData includes this option, we can start developing code generation based on `compare_by` in `base`.
 2. Add a convenience method for creating instances of `Just`.
 3. Use `protected` modifier with `Policy.whenever(..)` to comply with the contract method spec.

### CODE REVIEW

File: compiler/src/test/kotlin/io/spine/protodata/plugin/JustTest.kt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        val just = Just.just(validEvent)
+        assertThat(just)
+            .containsExactly(validEvent)
+    }
+}
+
+private val validEvent = ProjectCreated
+    .newBuilder()
+    .setId(newUuid())
+    .build()
```
Comment (yuri-sergiichuk) [inline compiler/src/test/kotlin/io/spine/protodata/plugin/JustTest.kt:54]: Why not `vBuild` or `buildPartial`?
```suggestion
private val validEvent = ProjectCreated.newBuilder()
    .setId(newUuid())
    .build()
```

**Review by yuri-sergiichuk — APPROVED** [review]

File: buildSrc/src/main/kotlin/io/spine/internal/gradle/Repositories.kt [hunk 2]
```diff
@@ -110,6 +115,44 @@ object PublishingRepos {
...
+     * Then, the key must be placed somewhere on the file system and the environment variable
+     * `GOOGLE_APPLICATION_CREDENTIALS` must be set to point at the key file.
+     * Once these preconditions are met, publishing becomes possible.
+     *
+     * ## Implementation note
+     * Google provides [tools](https://github.com/GoogleCloudPlatform/artifact-registry-maven-tools)
+     * for configuring authentication for the Maven repositories, including a Gradle plugin.
+     * However, the plugin is incompatible with Gradle 7.x at the moment.
+     * For now, we reproduce what the plugin does manually. This makes the whole `buildSrc`
+     * depend on the `artifactregistry-auth-common` artifact. Track [this issue](https://github.com/GoogleCloudPlatform/artifact-registry-maven-tools/issues/52)
```
Comment (alexander-yevsyukov) [inline buildSrc/src/main/kotlin/io/spine/internal/gradle/Repositories.kt:133]: Please have `Track this issue...`  coming as a next paragraph. And “‘please‘ would be nice”.

**Review by alexander-yevsyukov — APPROVED** [review]
LGTM, with a couple of minor doc-related suggestions.

File: compiler/src/test/kotlin/io/spine/protodata/plugin/JustTest.kt (hunk 1 above)
Comment (dmdashenkov) [reply compiler/src/test/kotlin/io/spine/protodata/plugin/JustTest.kt]: I've changed the method to `vBuild()` for now. However, please keep in mind that in future, `vBuild()`-s will die off and `build()` will validate messages.

**Pull request merged**